sky/packages/sky/lib/src/widgets/basic.dart

```diff
@@ -513,8 +513,60 @@ class BlockBody extends MultiChildRenderObjectWidget {
 }
 
 class Stack extends MultiChildRenderObjectWidget {
-  Stack(List<Widget> children, { Key key }) : super(key: key, children: children);
-  RenderStack createRenderObject() => new RenderStack();
+  Stack(List<Widget> children, {
+    Key key,
+    this.horizontalAlignment: 0.0,
+    this.verticalAlignment: 0.0
+  }) : super(key: key, children: children) {
+    assert(horizontalAlignment != null);
+    assert(verticalAlignment != null);
+  }
+
+  final double horizontalAlignment;
+  final double verticalAlignment;
+
+  RenderStack createRenderObject() {
+    return new RenderStack(
+      horizontalAlignment: horizontalAlignment,
+      verticalAlignment: verticalAlignment
+    );
+  }
+
+  void updateRenderObject(RenderStack renderObject, Stack oldWidget) {
+    renderObject.horizontalAlignment = horizontalAlignment;
+    renderObject.verticalAlignment = verticalAlignment;
+  }
+}
+
+class IndexedStack extends Stack {
+  IndexedStack(List<Widget> children, {
+    Key key,
+    double horizontalAlignment: 0.0,
+    double verticalAlignment: 0.0,
+    this.index: 0
+  }) : super(children,
```

Comment:
I didn't do this because the other similar classes in basic.dart are formatted this way and it looks OK IMHO.